### DISCUSSION TITLE
Tasmota Thermostatn persistent "TempTargetSet1" - Tasmota Rules

### DISCUSSION THREAD

**Discussion by DerT94** [arendst/Tasmota#20361] [Q&A]
Hello guys,
quick questions.
I am currently building a thermostat and want to make TempTargetSet1 persistent.
I can set the value with MQTT or TempTargetSet1 VALUE.
How can I create a rule that saves a change of the value to MEM1?

Something like

```
ON temptargetset1#change DO mem1 $value% ENDON
ON Power1#boot DO temptargetset1 DO 
```

Ther ist an Example but i don't get it.
https://tasmota.github.io/docs/Thermostat/#thermostat-persistent-storage-for-configuration

```
ON Power1#boot DO Backlog sensorinputset 1;controllermodeset 2;thermostatmodeset 1;temptargetset %mem1% ENDON
ON mqtt#connected DO Publish2 stat/TestTopic/targetTempValue {"Temp":%mem1%} ENDON
ON mem1#state DO Backlog temptargetset %value%;Publish2 stat/TestTopic/targetTempValue {"Temp":%mem1%} ENDON
```

Nowhere is mem1 set in this example

**Comment by sfromis** [comment]
The intention is that you use the command `Mem1` to set your preferred value, and then the rules will update the Thermostat feature with that value, after boot, or when the variable was changed by you.

**Reply by DerT94** [reply]
okay, that makes sense.
But I would like to use the last value after a restart.
My idea was to set mem1 on every change and read this value on boot

**Reply by sfromis** [reply]
Yes, that is exactly what happens when you use the `Mem1` command to perform the change.

**Reply by DerT94** [reply]
I don't get it
How do I set mem1 if the value behind temptargetset1 changes?
What condition do I need ?

**Comment by DerT94** [comment]
Got it
temptargetset1#Data DO mem1 %value% ENDON
